tools/containers/_common/pc-container-json-rpc.py

```
@@ -0,0 +1,116 @@
+    if not command:
+        raise PartcadJsonRpcException(-32602, "Command parameter is required")
+
+    # TODO(clairbee): input data validation for output files
+
+    # Replace the file names with temporary files
+    temp_files = []
+    for i in range(1, len(command)):
+        if not isinstance(command[i], str):
+            raise PartcadJsonRpcException(-32602, f"Command parameter at index {i} is not a string")
+        if command[i] in input_files:
+            temp_file = tempfile.NamedTemporaryFile(delete=True, suffix=os.path.splitext(command[i])[1])
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)